### ISSUE TITLE
Docker image build failed on local

### ISSUE THREAD

**Issue by hung-yueh** [AirenSoft/OvenMediaEngine#120]
Hi Team, thanks for the great work. 
I'm trying to build the docker image without any custom changes and got the build failure described below. Thanks for the help.

**Describe the bug**
Docker image build failed on local.

**To Reproduce**
1. git clone https://github.com/AirenSoft/OvenMediaEngine.git
2. `cd OvenMediaEngine`
3. `docker build --tag ovenmediaengine:debug . --no-cache`
4. see the failure below

**Expected behavior**
Should not fail

**Logs**
```
Step 19/26 : RUN         DIR=/tmp/ome &&         mkdir -p ${DIR} &&         cd ${DIR} &&         curl -sLf https://github.com/AirenSoft/OvenMediaEngine/archive/${OME_VERSION}.tar.gz | tar -xz --strip-components=1 &&         cd src &&         make release
 ---> Running in 7e817b4f91f3
####################################################
#                                                  #
#           AirenSoft Make System v0.63            #
#                                                  #
####################################################
   - Operating system        : linux
   - Core count              : 8
   - GCC version             : 7.5.0
   - Build option            : RELEASE
   - Paths for linker        : /opt/ovenmediaengine/lib:/opt/ovenmediaengine/lib64
   - Paths for pkg-config    : /opt/ovenmediaengine/lib/pkgconfig:/opt/ovenmediaengine/lib64/pkgconfig
   - Total Projects count    : 34

[  1/254|  0%] Preparing directories...
    Preparing directory bin/RELEASE...
[  2/254|  0%] [static] libapplication.a: C++ projects/base/info/logger_tag_info.cpp => intermediates/RELEASE/objs/base/info/logger_tag_info.o
    Preparing directory intermediates/RELEASE/static...
[  3/254|  1%] [static] libapplication.a: C++ projects/base/info/video_track.cpp => intermediates/RELEASE/objs/base/info/video_track.o
[ 10/254|  3%] [static] libovcrypto.a: C++ projects/base/ovcrypto/base_64.cpp => intermediates/RELEASE/objs/base/ovcrypto/base_64.o
[  9/254|  3%] [static] libapplication.a: C++ projects/base/info/media_track.cpp => intermediates/RELEASE/objs/base/info/media_track.o
[ 15/254|  5%] [static] libovcrypto.a: C++ projects/base/ovcrypto/openssl/tls_data.cpp => intermediates/RELEASE/objs/base/ovcrypto/openssl/tls_data.o
[  4/254|  1%] [static] libapplication.a: C++ projects/base/info/host.cpp => intermediates/RELEASE/objs/base/info/host.o
[  8/254|  3%] [static] libapplication.a: C++ projects/base/info/application.cpp => intermediates/RELEASE/objs/base/info/application.o
[  6/254|  2%] [static] libapplication.a: C++ projects/base/info/audio_track.cpp => intermediates/RELEASE/objs/base/info/audio_track.o
[  7/254|  2%] [static] libapplication.a: C++ projects/base/info/stream.cpp => intermediates/RELEASE/objs/base/info/stream.o
[  5/254|  1%] [static] libapplication.a: C++ projects/base/info/session.cpp => intermediates/RELEASE/objs/base/info/session.o
[ 14/254|  5%] [static] libovcrypto.a: C++ projects/base/ovcrypto/openssl/openssl_manager.cpp => intermediates/RELEASE/objs/base/ovcrypto/openssl/openssl_manager.o
[ 12/254|  4%] [static] libovcrypto.a: C++ projects/base/ovcrypto/crc_32.cpp => intermediates/RELEASE/objs/base/ovcrypto/crc_32.o
[ 13/254|  5%] [static] libovcrypto.a: C++ projects/base/ovcrypto/message_digest.cpp => intermediates/RELEASE/objs/base/ovcrypto/message_digest.o
[ 11/254|  4%] [static] libovcrypto.a: C++ projects/base/ovcrypto/certificate.cpp => intermediates/RELEASE/objs/base/ovcrypto/certificate.o
[ 16/254|  6%] [static] libovcrypto.a: C++ projects/base/ovcrypto/openssl/tls.cpp => intermediates/RELEASE/objs/base/ovcrypto/openssl/tls.o
[ 17/254|  6%] [static] libovlibrary.a: C++ projects/base/ovlibrary/random.cpp => intermediates/RELEASE/objs/base/ovlibrary/random.o
[ 18/254|  7%] [static] libovlibrary.a: C++ projects/base/ovlibrary/json_object.cpp => intermediates/RELEASE/objs/base/ovlibrary/json_object.o
[ 20/254|  7%] [static] libovlibrary.a: C++ projects/base/ovlibrary/assert.cpp => intermediates/RELEASE/objs/base/ovlibrary/assert.o
[ 19/254|  7%] [static] libovlibrary.a: C++ projects/base/ovlibrary/data.cpp => intermediates/RELEASE/objs/base/ovlibrary/data.o
[ 21/254|  8%] [static] libovlibrary.a: C++ projects/base/ovlibrary/semaphore.cpp => intermediates/RELEASE/objs/base/ovlibrary/semaphore.o
[ 22/254|  8%] [static] libovlibrary.a: C++ projects/base/ovlibrary/stack_trace.cpp => intermediates/RELEASE/objs/base/ovlibrary/stack_trace.o
[ 23/254|  9%] [static] libovlibrary.a: C++ projects/base/ovlibrary/platform.cpp => intermediates/RELEASE/objs/base/ovlibrary/platform.o
[ 24/254|  9%] [static] libovlibrary.a: C++ projects/base/ovlibrary/pcm_utilities.cpp => intermediates/RELEASE/objs/base/ovlibrary/pcm_utilities.o
[ 25/254|  9%] [static] libovlibrary.a: C++ projects/base/ovlibrary/path_manager.cpp => intermediates/RELEASE/objs/base/ovlibrary/path_manager.o
[ 26/254| 10%] [static] libovlibrary.a: C++ projects/base/ovlibrary/error.cpp => intermediates/RELEASE/objs/base/ovlibrary/error.o
[ 28/254| 11%] [static] libovlibrary.a: C++ projects/base/ovlibrary/delay_queue.cpp => intermediates/RELEASE/objs/base/ovlibrary/delay_queue.o
[ 27/254| 10%] [static] libovlibrary.a: C++ projects/base/ovlibrary/string.cpp => intermediates/RELEASE/objs/base/ovlibrary/string.o
[ 29/254| 11%] [static] libovlibrary.a: C++ projects/base/ovlibrary/json.cpp => intermediates/RELEASE/objs/base/ovlibrary/json.o
[ 30/254| 11%] [static] libovlibrary.a: C++ projects/base/ovlibrary/byte_stream.cpp => intermediates/RELEASE/objs/base/ovlibrary/byte_stream.o
[ 31/254| 12%] [static] libovlibrary.a: C++ projects/base/ovlibrary/stop_watch.cpp => intermediates/RELEASE/objs/base/ovlibrary/stop_watch.o
[ 32/254| 12%] [static] libovlibrary.a: C++ projects/base/ovlibrary/queue.cpp => intermediates/RELEASE/objs/base/ovlibrary/queue.o
[ 33/254| 12%] [static] libovlibrary.a: C++ projects/base/ovlibrary/dump_utilities.cpp => intermediates/RELEASE/objs/base/ovlibrary/dump_utilities.o
[ 35/254| 13%] [static] libovlibrary.a: C++ projects/base/ovlibrary/log_write.cpp => intermediates/RELEASE/objs/base/ovlibrary/log_write.o
[ 34/254| 13%] [static] libovlibrary.a: C++ projects/base/ovlibrary/log.cpp => intermediates/RELEASE/objs/base/ovlibrary/log.o
[ 36/254| 14%] [static] libovlibrary.a: C++ projects/base/ovlibrary/daemon.cpp => intermediates/RELEASE/objs/base/ovlibrary/daemon.o
[ 38/254| 14%] [static] libovlibrary.a: C++ projects/base/ovlibrary/url.cpp => intermediates/RELEASE/objs/base/ovlibrary/url.o
[ 37/254| 14%] [static] libovlibrary.a: C++ projects/base/ovlibrary/event.cpp => intermediates/RELEASE/objs/base/ovlibrary/event.o
[ 39/254| 15%] [static] libovlibrary.a: C++ projects/base/ovlibrary/stl.cpp => intermediates/RELEASE/objs/base/ovlibrary/stl.o
[ 40/254| 15%] [static] libovlibrary.a: C++ projects/base/ovlibrary/log_internal.cpp => intermediates/RELEASE/objs/base/ovlibrary/log_internal.o
[ 41/254| 16%] [static] libsocket.a: C++ projects/base/ovsocket/socket_address.cpp => intermediates/RELEASE/objs/base/ovsocket/socket_address.o
[ 42/254| 16%] [static] libsocket.a: C++ projects/base/ovsocket/socket_utilities.cpp => intermediates/RELEASE/objs/base/ovsocket/socket_utilities.o
[ 43/254| 16%] [static] libsocket.a: C++ projects/base/ovsocket/socket.cpp => intermediates/RELEASE/objs/base/ovsocket/socket.o
[ 44/254| 17%] [static] libsocket.a: C++ projects/base/ovsocket/client_socket.cpp => intermediates/RELEASE/objs/base/ovsocket/client_socket.o
[ 45/254| 17%] [static] libsocket.a: C++ projects/base/ovsocket/server_socket.cpp => intermediates/RELEASE/objs/base/ovsocket/server_socket.o
[ 47/254| 18%] [static] libprovider.a: C++ projects/base/provider/stream.cpp => intermediates/RELEASE/objs/base/provider/stream.o
[ 46/254| 18%] [static] libsocket.a: C++ projects/base/ovsocket/datagram_socket.cpp => intermediates/RELEASE/objs/base/ovsocket/datagram_socket.o
[ 48/254| 18%] [static] libprovider.a: C++ projects/base/provider/application.cpp => intermediates/RELEASE/objs/base/provider/application.o
[ 49/254| 19%] [static] libprovider.a: C++ projects/base/provider/provider.cpp => intermediates/RELEASE/objs/base/provider/provider.o
[ 50/254| 19%] [static] libpublisher.a: C++ projects/base/publisher/publisher.cpp => intermediates/RELEASE/objs/base/publisher/publisher.o
g++: internal compiler error: Killed (program cc1plus)
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
make: *** [intermediates/RELEASE/objs/base/info/stream.o] Error 4
make: *** Waiting for unfinished jobs....
core/build_objects.mk:18: recipe for target 'intermediates/RELEASE/objs/base/info/stream.o' failed
core/build_objects.mk:18: recipe for target 'intermediates/RELEASE/objs/base/info/application.o' failed
g++: internal compiler error: Killed (program cc1plus)
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
make: *** [intermediates/RELEASE/objs/base/info/application.o] Error 4
core/build_objects.mk:18: recipe for target 'intermediates/RELEASE/objs/base/info/host.o' failed
g++: internal compiler error: Killed (program cc1plus)
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
make: *** [intermediates/RELEASE/objs/base/info/host.o] Error 4
The command '/bin/sh -c DIR=/tmp/ome &&         mkdir -p ${DIR} &&         cd ${DIR} &&         curl -sLf https://github.com/AirenSoft/OvenMediaEngine/archive/${OME_VERSION}.tar.gz | tar -xz --strip-components=1 &&         cd src &&         make release' returned a non-zero code: 2
```

**Server (please complete the following information):**
 - OS: Docker Desktop runs on Mac
 - OvenMediaEngine Version: v0.10.3
 - Branch: master

**Comment by getroot** [comment]
We are not reproducing this problem.
We are using Ubuntu to build Docker images. Currently, we can't use a Mac to build Docker images, so it can take some time. Can you use the pre-built version?

https://hub.docker.com/repository/docker/airensoft/ovenmediaengine

**Comment by hung-yueh** [comment]
Hi @getroot, thanks for getting back to me.

I'm actually using Docker to build the image which is running on top of Ubuntu 18.04 that you specified in the Dockerfile. The host machine(MacOS) shouldn't matter and that's the whole point to use the container technology, please correct me if it's incorrect for OME. I'm also happy to help to debug if it should work, please let me know.

The above log shows the information: 
```
####################################################
#                                                  #
#           AirenSoft Make System v0.63            #
#                                                  #
####################################################
   - Operating system        : linux
   - Core count              : 8
   - GCC version             : 7.5.0
   - Build option            : RELEASE
```
You can see it's building on Linux, not MacOS.

One question I found here is: 
Does OME have any GCC version requirements?

**Comment by hung-yueh** [comment]
@getroot Thanks for getting back to me.

**Comment by getroot** [comment]
I tried building the Dockerfile of the Master branch once again today. But the problem is not reproduced. Do you know the cause? Is there any difference in the Docker build system on Mac?

**Comment by hung-yueh** [comment]
@getroot Thanks for following this up. Docker builds a system environment based on the Dockerfile, and that environment builds the OME code. The host machine(macOS in this case) that runs the Docker shouldn't matter unless there are bugs in the Docker's code. I have found the root cause is that the memory resource that I allocated to Docker was not enough to build the OME image. It surprisingly needs more than 5GB memory. My current config is 8 GB to build the image without the problem, you probably can add this into README or the document to help others.
